IndexError: list index out of range error during Windows Server 2019 and 2022 deployment

| Version | Revision | Production/Development | Component | Install type |
|:--:|:--:|:--:|:--:|:--:|
|  v4.9.2-rc1  |  1  |  Development  |  Allocation module  | - | 



## Description


During the tests performed at https://github.com/wazuh/wazuh/issues/26614, the deployment of Windows Server 2019 and 2022 hosts failed. 
The following log could be observed:


```[31m[2024-10-29 14:47:23] [ERROR] [477704] [ThreadPoolExecutor-0_7] [jobflow]: [allocate-windows-agent-windows-server-2019-amd64] Task failed with error: Error executing process task Traceback (most recent call last):
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/main.py", line 39, in <module>
    main()
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/main.py", line 35, in main
    Allocator.run(InputPayload(**vars(parse_arguments())))
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/allocation.py", line 40, in run
    return cls.__create(payload)
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/allocation.py", line 59, in __create
    instance = provider.create_instance(
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/generic/provider.py", line 71, in create_instance
    return cls._create_instance(base_dir, params, config, ssh_key)
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/aws/provider.py", line 119, in _create_instance
    instance_id = cls.__create_ec2_instance(config)
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/aws/provider.py", line 204, in __create_ec2_instance
    root_device_name = ami['Images'][0]['RootDeviceName']
IndexError: list index out of range
[0m
[32m[2024-10-29 14:47:23] [INFO] [477704] [ThreadPoolExecutor-0_7] [jobflow]: [allocate-windows-agent-windows-server-2022-amd64] Starting task.[0m
[36m[2024-10-29 14:47:23] [DEBUG] [477704] [ThreadPoolExecutor-0_7] [jobflow]: Running task "allocate-windows-agent-windows-server-2022-amd64" with arguments: ['modules/allocation/main.py', '--action=create', '--provider=aws', '--size=large', '--composite-name=windows-server-2022-amd64', '--inventory-output=/tmp/dtt1-poc/agent-windows-server-2022-amd64/inventory.yaml', '--track-output=/tmp/dtt1-poc/agent-windows-server-2022-amd64/track.yaml', '--label-termination-date=1d', '--label-team=qa', '--label-issue=https://github.com/wazuh/wazuh/issues/26614'][0m
[31m[2024-10-29 14:47:25] [ERROR] [477704] [ThreadPoolExecutor-0_7] [jobflow]: [allocate-windows-agent-windows-server-2022-amd64] Task failed with error: Error executing process task Traceback (most recent call last):
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/main.py", line 39, in <module>
    main()
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/main.py", line 35, in main
    Allocator.run(InputPayload(**vars(parse_arguments())))
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/allocation.py", line 40, in run
    return cls.__create(payload)
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/allocation.py", line 59, in __create
    instance = provider.create_instance(
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/generic/provider.py", line 71, in create_instance
    return cls._create_instance(base_dir, params, config, ssh_key)
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/aws/provider.py", line 119, in _create_instance
    instance_id = cls.__create_ec2_instance(config)
  File "/home/akim/Desktop/branch_management/wazuh-qa/deployability/modules/allocation/aws/provider.py", line 204, in __create_ec2_instance
    root_device_name = ami['Images'][0]['RootDeviceName']
IndexError: list index out of range
```



### Environment

branch v4.9.2-rc1

### Steps to reproduce


1. Clone wazuh-qa repository

2. Move to v4.9.2-rc1 branch

3. Follow the README in deployability

4. Use the following YAML:  [Test-agents-complete.yaml.txt](https://github.com/user-attachments/files/17556908/Test-agents-complete.yaml.txt)


### Current result

Deployment failure

### Expected result

No errors during deployment were expected

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,7 +6,7 @@ All notable changes to this project will be documented in this file.
 
 ### Fixed
 
-- Updated Windows server AMis. ([#5831](https://github.com/wazuh/wazuh-qa/pull/5831))
+- Updated Windows server AMis. ([#5831](https://github.com/wazuh/wazuh-qa/pull/5831)) \- (Framework)
 
 ## [4.9.1] - TBD
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file.
 
 ## [4.9.2] - TBD
 
+### Fixed
+
+- Updated Windows server AMis. ([#5831](https://github.com/wazuh/wazuh-qa/pull/5831))
+
 ## [4.9.1] - TBD
 
 ### Added

--- a/deployability/modules/allocation/static/specs/os.yml
+++ b/deployability/modules/allocation/static/specs/os.yml
@@ -463,14 +463,14 @@ aws:
     zone: us-east-1
     user: wazuh-user
   windows-server-2016-amd64:
-    ami: ami-0f279f6c0764bef8f
+    ami: ami-0d0ae05c222fd12ec
     zone: us-east-1
     user: wazuh-user
   windows-server-2019-amd64:
-    ami: ami-06cc514f1012a7431
+    ami: ami-073e3b46f8802d31b
     zone: us-east-1
     user: wazuh-user
   windows-server-2022-amd64:
-    ami: ami-0f9c44e98edf38a2b
+    ami: ami-0324a83b82023f0b3
     zone: us-east-1
     user: wazuh-user

